polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistTypes.td

```diff
@@ -31,18 +31,31 @@ def Polygeist_StructType :
   let description = [{
     Type used to represent a collection of data members together in memory.
   }];
-  let parameters = (ins ArrayRefParameter<"mlir::Type">:$body,
+  let parameters = (ins OptionalArrayRefParameter<"mlir::Type">:$body,
                         OptionalParameter<"std::optional<StringAttr>">:$name,
-                        DefaultValuedParameter<"bool", "false">:$isPacked);
+                        DefaultValuedParameter<"bool", "false">:$isPacked,
+                        DefaultValuedParameter<"bool", "false">:$isOpaque);
   let builders = [
     TypeBuilder<(ins "llvm::ArrayRef<mlir::Type>":$body),[{
-      return $_get($_ctxt, body, std::nullopt, false);
+      return $_get($_ctxt, body, std::nullopt, false, false);
+    }]>,
+    TypeBuilder<(ins "llvm::ArrayRef<mlir::Type>":$body, "std::optional<StringAttr>":$name, "bool":$isPacked),[{
+      return $_get($_ctxt, body, name, isPacked, false);
     }]>
   ];
   let extraClassDeclaration = [{
     bool isPacked() { return getIsPacked(); }
+    bool isOpaque() { return getIsOpaque(); }
+    static StructType getOpaque(StringAttr name, mlir::MLIRContext *context) {
```

Comment:
Is it possible to construct an opaque `polygeist.struct` type through the default builders? 

If so, do we verify that any opaque struct has a name?

Reply:
Maybe it makes sense to add a verifier checking struct types either have a name or a body.

Reply:
Added verifier to check opaque struct has a name.
The default builder takes body as `ArrayRef `, so it cannot be nullopt.

Reply:
`ArrafRef` is constructible from `nullopt`: https://llvm.org/doxygen/classllvm_1_1ArrayRef.html#aa1459c8bf806db60a9a1c2d11f95536d